src/components/NavigationBar/NavigationBar.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { NavigationBar } from "./NavigationBar";
+import { BrowserRouter, Route } from "react-router-dom";
+import rinkebyConfig from "../../test/fixtures/sample-rinkeby-config.json";
+import ropstenConfig from "../../test/fixtures/sample-config.json";
+
+const mockMainnet = {
+  network: "homestead",
+};
+
+describe("navigationBar", () => {
+  it("should render the UI accordingly", () => {
+    render(
+      <BrowserRouter>
+        <Route exact path="/">
+          <NavigationBar logout={() => {}} />
+        </Route>
+      </BrowserRouter>
+    );
+
+    expect(screen.queryAllByText("Verify Documents")).not.toBeNull();
+    expect(screen.queryAllByText("Create Documents")).not.toBeNull();
+    expect(screen.queryAllByText("Resources")).not.toBeNull();
+    expect(screen.queryAllByText("FAQ")).not.toBeNull();
+    expect(screen.queryAllByText("Contact")).not.toBeNull();
+    expect(screen.queryByTestId("settings-icon")).not.toBeNull();
+  });
+
+  it("should go to default verify documents page without config", () => {
+    render(
+      <BrowserRouter>
+        <Route exact path="/">
+          <NavigationBar logout={() => {}} />
+        </Route>
+      </BrowserRouter>
+    );
+
+    expect(screen.getByText("Verify Documents").closest("a")).toHaveAttribute(
+      "href",
+      "https://tradetrust.io/#verify-documents"
+    );
```

Comment:
so it's `queryByText` will do:

```
expect(screen.queryByText("Verify Documents")).toHaveAttribute(
  "href",
  "https://tradetrust.io/#verify-documents"
);
```

Reply:
Based on: [this link](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence) might be better to keep as `getByText` for an error to be thrown if it does not exist.

<img width="950" alt="Screenshot 2020-11-02 at 5 02 11 PM" src="https://user-images.githubusercontent.com/38589870/97849199-2e276680-1d2d-11eb-8acf-a4d100d5a837.png">

Reply:
ah nice, then how about:

```
expect(screen.getByText("Verify Documents").getAttribute("href")).toBe(
  "https://tradetrust.io/#verify-documents"
);
```

your solution works fine actually, just that i feel maybe can try to omit `.closest("a")`, cos there is an assumption/opinion on html markup structure when it is doing a find `closest` anchor tag, we'll want to keep it as less opinionated as possible